From fd3af89c3193bd05e0924e6bc7add6ffbb4ffe8d Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 6 Sep 2016 19:54:08 +0100
Subject: [PATCH] Schedule update in CRepositoryUpdater by triggering the
 setting change callback but set to current value so we respect user settings

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/Application.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 9ed74f1..6c0ef01 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -3012,7 +3012,7 @@ void CApplication::Stop(int exitCode)
 void CApplication::SetOSMCWalkthroughState(OSMCWalkthroughState state)
 {
  m_eOSMCWalkthroughState = state;
- CRepositoryUpdater::GetInstance().ScheduleUpdate();
+    CSettings::GetInstance().SetInt(CSettings::SETTING_ADDONS_AUTOUPDATES, CSettings::GetInstance().GetInt(CSettings::SETTING_ADDONS_AUTOUPDATES));
 }
 
 bool CApplication::PlayMedia(const CFileItem& item, const std::string &player, int iPlaylist)
-- 
2.7.4

